core/http/src/uri/absolute.rs

```diff
@@ -115,6 +129,26 @@ impl<'a> Absolute<'a> {
         crate::parse::uri::absolute_from_str(string)
     }
 
+    /// Parses the string `string` into an `Absolute`. Parsing will never
+    /// allocate. This method should be used instead of
+    /// [`Absolute::parse()`](crate::uri::Absolute::parse()) when the source URI is
+    /// already a `String`. Returns an `Error` if `string` is not a valid absolute
+    /// URI.
+    pub fn parse_owned(string: String) -> Result<Absolute<'static>, Error<'static>> {
+        let absolute = Absolute::parse(&string).map_err(|e| e.into_owned())?;
+        debug_assert!(absolute.source.is_some(), "Origin source parsed w/o source");
+
+        let absolute = Absolute {
+            scheme: absolute.scheme.into_owned(),
+            authority: absolute.authority.into_owned(),
```

Comment:
This will allocate. This is why the method was not implemented for this type. I'm not quite sure what to do about this.